### DISCUSSION TITLE
Nefungující ssh klíč - Error loading key "(stdin)": invalid format

### DISCUSSION THREAD

**Discussion by marekvesely-direct** [canarytrace/documentation#182] [Synthetic] [answered]
Dobrý den,

postupuji podle návodu na vytvoření ssh klíče (https://canarytrace.com/docs/guides/wdio/#run-canarytrace-in-kubernetes), v command line si přes klíč daný repozitář naklonuju, ale přes Kubernetes mi to píše chybu viy obrázek. Klíč jsem vytvořil ve formátu jak to máte v dokumentaci: ssh-keygen -t ed25519 -C "tajne@direct.cz". Můžete poradit co s tím?

Děkuji,

Veselý

![a](https://user-images.githubusercontent.com/56677711/213101554-0998adc0-7223-4e0b-8609-03377e4a213d.PNG)

**Comment by rdpanek** [comment, marked as answer]
Dobrý den Marku, zkuste to ještě jednou podle tohoto: https://github.com/canarytrace/documentation/discussions/178 Nedávno proběhlo právě testování a vše bylo OK. Díky

**Reply by marekvesely-direct** [reply]
@rdpanek Je to stejný postup a taky nefunguje, máme tuto image: canarytrace:4.22.0-devops-20221121083205-28, nemůže být problém v té image? Případně pošlete mi na email marek.vesely@direct.cz kde stáhnout aktuální funkční verzi, ať si ji naklonujeme k sobě. Děkuji

**Reply by rdpanek** [reply]
1). Máte latest build, spíše zkuste přijít na to, proč Vám ten klíč hlásí `invalid format`, řeší to tato část kódu
```bash
            echo "Use ssh key"
            SSH_KEY="/secret/ssh-privatekey"

            # ssh agent and key available in env
            ls -lah $SSH_KEY
            eval $(ssh-agent)
            ssh-add - < ${SSH_KEY}

            # make git and whoami not complain about unknown user
            ls -l /etc/passwd
            whoami
```

2). Druhý způsob, jak na problém přijít je použít docker image lokálně, např. `--entrypoint=/bin/sh` a tak můžete debugovat přímo v běžícím kontejneru. Entrypoint, který se jinak při startu spouští je `/opt/canary/entrypoint.sh` a samozřejmě, klíč do kontejneru dostanete přes přepínač `-v $(pwd)`.

**Reply by rdpanek** [reply]
S největší pravděpodobností bude problém v příkazu `ssh-add - < ${SSH_KEY}` kdy se mu nelíbí formát, například přehození public a private. Také zkontrolujte, jestli máte klíč naimportovaný v secret správně.

**Reply by marekvesely-direct** [reply]
Děkuji za pomoc s debugem, problém byl, že jsem měl privátní klíč bez prázdného řádku na konci, když jsem přidal prázdný řádek na konec privátního klíče tak vše jede.

**Reply by rdpanek** [reply]
tvl mám radost, dobrá práce 🤟
